playbooks/install-ambari-2_6.yml

```diff
@@ -1,20 +1,26 @@
-- hosts: ambari-server
+- hosts: ambari-server ambari-agent
   become: true
   vars:
-    ambari_base_version: 2.x
     ambari_version: 2.6.0.0
+    ambari_base_version: "{{ ambari_version[0] }}.x"
+  tasks:
+    - include_vars: file="../vars/{{ ansible_distribution }}.yml"
+    - include_vars: file="../roles/ambari-common/vars/ambari-public-repo-root.yml"
+      when: ambari_repo_root is not defined
+    - include_vars: file="../roles/ambari-common/vars/ambari-public-repo-base-url.yml"
+      when: ambari_repo_base_url is not defined
+    - include_tasks: ../roles/ambari-common/tasks/install-repo-{{ ansible_os_family }}.yml
+
+- hosts: ambari-server
+  become: true
   tasks:
-    - include: ../roles/ambari-common/tasks/install-dev-yum-repo.yml
-    - include: ../roles/ambari-server/tasks/install.yml
-    - include: ../roles/ambari-server/tasks/setup.yml
-    - include: ../roles/ambari-server/tasks/start.yml
+    - include_tasks: ../roles/ambari-server/tasks/install-{{ ansible_os_family }}.yml
+    - import_tasks: ../roles/ambari-server/tasks/setup.yml
+    - import_tasks: ../roles/ambari-server/tasks/start.yml
+
 - hosts: ambari-agent
   become: true
-  vars:
-    ambari_base_version: 2.x
-    ambari_version: 2.6.0.0
   tasks:
-    - include: ../roles/ambari-common/tasks/install-dev-yum-repo.yml
-    - include: ../roles/ambari-agent/tasks/install.yml
-    - include: ../roles/ambari-agent/tasks/setup.yml
-    - include: ../roles/ambari-agent/tasks/start.yml
+    - include_tasks: ../roles/ambari-agent/tasks/install-{{ ansible_os_family }}.yml
```

Comment:
thanks for the change, regarding to include -> include_tasks transition, there is a better solution that maybe you can try or use:

include_role:
    name: ambari-agent
    tasks_from:  "install-{{ ansible_os_family }}"

otherwise, the changes are fine

Reply:
Doesn't seem to work when task name includes variables.  Issue is reported at ansible/ansible#22571 (and marked as dup of ansible/ansible#22881).

Reply:
ok, then if you feel its fine, then i can merge the PR, later we can cleanup and use include_role when the issues with that will be fixed